app/models/grda_warehouse/warehouse_reports/confidential_touch_point.rb

```diff
@@ -0,0 +1,66 @@
+###
+# Copyright 2016 - 2020 Green River Data Analysis, LLC
+#
+# License detail: https://github.com/greenriver/hmis-warehouse/blob/master/LICENSE.md
+###
+
+module GrdaWarehouse::WarehouseReports
+  class ConfidentialTouchPoint < TouchPoint
+
+    def title
+      'Confidential Touch Point Export'
+    end
+
+    def url
+      warehouse_reports_confidential_touch_point_exports_url(host: ENV.fetch('FQDN'))
+    end
+
+    # Convert the complex data structure to an array of arrays
+    # for easier storage and Excel export
+    def clean_data(dirty)
+      @cleaned_data ||= begin
+        clean = []
+        section_columns = sections(dirty).map do |title, questions|
+          column_count = questions.size - 1
+          column_count = 0 if column_count < 0
+          [title] + [nil]*(column_count)
+        end
+        clean << ['Selected Range:', start_date, end_date] + section_columns.flatten
+        clean << ["Client ID", 'Medicaid ID', "Client Name", "Collected On", "Location", 'SDH Enroll Date', "Staff"] + all_questions(dirty)
```

Comment:
I'd rather see these two lines as separate items in a hash or struct than stuffed into the beginning of the array. That way the array of arrays contains the cleaned data, and the headers are headers.

Reply:
We could keep them separate, but because many of the first are empty and the placement is important, it's probably safer to leave them as arrays.  These end up essentially being a huge CSV with two header rows.